mesonbuild/environment.py

```diff
@@ -1016,6 +1016,37 @@ def detect_compilers(self, lang, need_cross_compiler):
                 raise EnvironmentException('Cross compilation with Swift is not working yet.')
                 # cross_comp = self.environment.detect_fortran_compiler(True)
 
+        if comp is None:
+            if not introspection_mode:
+                raise EnvironmentException('Tried to use unknown language "%s".' % lang)
+            else:
+                return None, None
+
```

Comment:
If we split right here to make a new function, the caller can

1. Call the old function, which is what was already in `environment.py`
2. Do the abort themselves, as before
3. Do the `sanity_check` calls, if applicable
4. Call the new function

Reply:
Sure, but I split it a bit differently: `detect_compilers` now calls both the old and the new function. The user can then decide to call the additional `check_compilers` function if necessary. This approach reduces the amount of code duplication and extra logic in the interpreter and introspector.

Reply:
Even better!